src/Octoshift/AdoApi.cs

```diff
@@ -431,7 +431,11 @@ public virtual async Task<int> GetPipelineId(string org, string teamProject, str
                 }
             }
 
-            return _pipelineIds[(org.ToUpper(), teamProject.ToUpper(), pipelinePath.ToUpper())];
+            return _pipelineIds.TryGetValue((org.ToUpper(), teamProject.ToUpper(), pipelinePath.ToUpper()), out result)
```

Comment:
Maybe I am missing something here but I think the `TryGetValue` here is redundant since if we reach this far it means that we have already done the `TryGetValue` earlier in line 413

Reply:
if line 413 had a value we would have returned on line 415. We need to TryGet again because we could have added a new value on line 426.

Reply:
The login in this method is basically 1. Do we have the ID in the cache? 2. Lets get all the values from the API, and lookup by path 3. Didn't find it? lets look up by name only 4. Still didn't find it, or found more than one? error